org.eclipse.m2e.jdt/src/org/eclipse/m2e/jdt/internal/AbstractJavaProjectConfigurator.java

```diff
@@ -198,6 +201,24 @@ public void configure(ProjectConfigurationRequest request, IProgressMonitor moni
     MavenJdtPlugin.getDefault().getBuildpathManager().updateClasspath(project, monitor);
   }
 
+  static String getHighestEnvironmentId(String... environmentIds) {
```

Comment:
Where could I put a unit test for this (no IT necessary, no bundling into an Eclipse plugin either)? m2e.jdt.tests also only contains package `o.e.m.jdt.tests` and this test would need to be located in the same package.

Reply:
The approach from https://stackoverflow.com/a/44926613 seems reasonable to me.

Reply:
o.e.m.jdt.tests would be the best place. As the class is internal you could make the method static.
Another approach, that m2e is not leveraging yet so it may require more work to get started, is https://github.com/eclipse/tycho/blob/master/RELEASE_NOTES.md#execute-unit-tests-with-eclipse-plugin-packaging